core/src/main/java/com/google/cloud/sql/CredentialFactory.java

```diff
@@ -26,4 +32,55 @@ public interface CredentialFactory {
   String CREDENTIAL_FACTORY_PROPERTY = "cloudSql.socketFactory.credentialFactory";
 
   HttpRequestInitializer create();
+
+  /**
+   * Returns a GoogleCredentials to use for generating tokens and authenticating web requests. The
+   * default implementation will attempt to extract the GoogleCredentials object from known
+   * credential HttpRequestInitializer implementations already in used to provide tokens for IAM
+   * Authentication.
+   */
+  default GoogleCredentials getCredentials() {
```

Comment:
Can we put some unit tests in place for this class? I'd like to see that the interface works before this method was added, and a test with it in use.

Reply:
Sure. Existing tests are in DefaultAccessTokenSupplierTest. I'll move them into a test class specifically for this interface.